nvc/toc.go

```diff
@@ -0,0 +1,53 @@
+package nvc
+
+import (
+	"io"
+	"encoding/binary"
+)
+
+// Reads the Table of Contents from r.
+// The ToC format is as follows:
+//   1. 32 LE unsigned integer -> number of entries in ToC.
+//   2. This repeated as many times as the last value indicates:
+//     a. FNV1a hash (64-bit LE unsigned integer)
+//     b. Offset in the NVC archive (64-bit LE unsigned integer)
+//     c. Uncompressed length of the Entry (64-bit LE unsigned integer)
+//     d. Actual length of the Entry (64-bit LE unsigned integer)
+//     e. Entry flags (64-bit LE unsigned integer)
+func ReadToc(r io.Reader) ([]TocEntry, error) {
+	count, countErr := readCount(r)
+	if countErr != nil {
+		return nil, countErr
+	}
```

Comment:
I think checking the magic bytes should be moved to the start of this function (so that we can then remove that check from everywhere else).

Right now, `ListEntries` is just "check for magic bytes then call this function".  `extractNVC` also has some code that....checks the magic bytes and then calls this function. I don't think any caller that just wants a list of ToC entries should have to worry about something as low-level as magic bytes.

To be clear, the change I'm proposing would include deleting the `ListEntries` function.

Reply:
:boom: